sympy/physics/control/tests/test_lti.py

```diff
@@ -0,0 +1,233 @@
+
+    # purely symbolic polynomials.
+    tf5 = TransferFunction(p, p)
+    assert tf5.num == tf5.den == p
+    assert tf5.args == (p, p)
+
+    tf6 = TransferFunction(5, 6)
+    assert tf6.num == 5
+    assert tf6.den == 6
+    assert tf6.args == (5, 6)
+
+    # ValueError when denominator is zero.
+    raises(ValueError, lambda: TransferFunction(4, 0))
+    raises(ValueError, lambda: TransferFunction(s, 0))
+    raises(ValueError, lambda: TransferFunction(0, 0))
+
+
+def test_TransferFunction_functions():
+    # explicitly cancel poles and zeros.
+    tf0 = TransferFunction(s**5 + s**3 + s, s - s**2)
+    a = TransferFunction(-s**4 - s**2 - 1, s - 1)
+    assert tf0.cancel_poles_and_zeros() == a
+
+    tf1 = TransferFunction((p + 3)*(p - 1), (p - 1)*(p + 5))
+    b = TransferFunction(p + 3, p + 5)
+    assert tf1.cancel_poles_and_zeros() == b
+
+    # negation of TF.
+    tf2 = TransferFunction(s + 3, s**2 - s**3 + 9)
+    tf3 = TransferFunction(-3*p + 3, 1 - p)
+    assert tf2.__neg__() == TransferFunction(-s - 3, s**2 - s**3 + 9)
+    assert tf3.__neg__() == TransferFunction(3*p - 3, 1 - p)
+
+    # taking power of a TF.
+    tf4 = TransferFunction(p + 4, p - 3)
+    tf5 = TransferFunction(s**2 + 1, 1 - s)
+    result = TransferFunction(s**6 + 3*s**4 + 3*s**2 + 1, -s**3 + 3*s**2 - 3*s + 1)
+    assert tf4.pow(2) == TransferFunction(p**2 + 8*p + 16, p**2 - 6*p + 9)
+    assert tf5.pow(3) == result
+
+    # sympy's own functions.
+    assert tf3.xreplace({p: s}) == TransferFunction(-3*s + 3, 1 - s)
+    tf = TransferFunction(s - 1, s**2 - 2*s + 1)
+    assert factor(tf) == TransferFunction(s - 1, (s - 1)**2)
+
+
+def test_TransferFunction_addition():
+    tf1 = TransferFunction(s + 6, s - 5)
+    tf2 = TransferFunction(s + 3, s + 1)
+    tf3 = TransferFunction(s + 1, s**2 + s + 1)
+    tf4 = TransferFunction(p, 2 - p)
+    assert tf1.add(tf2) == TransferFunction(2*s**2 + 5*s - 9, s**2 - 4*s - 5)
+    assert tf3.add(tf4) == \
+        TransferFunction(p*s**2 + 2*s + 2, -p*s**2 - p*s - p**2 + 4*s + 2)
+    assert tf1.add(s - 1) == TransferFunction(s**2 - 5*s + 11, s - 5)
+    assert tf1.add(8) == TransferFunction(9*s - 34, s - 5)
+
+
+def test_TransferFunction_multiplication_and_division():
+    G1 = TransferFunction(s + 3, -s**3 + 9)
+    G2 = TransferFunction(s + 1, s - 5)
+    expect = TransferFunction(-s**2 - 4*s - 3, s**4 - 5*s**3 -9*s + 45)
+    assert G1.mul(G2) == expect
+
+    G3 = TransferFunction(p, p**4 - 6)
+    G4 = TransferFunction(p + 4, p - 5)
+    expect = TransferFunction(p**2 + 4*p, p**5 - 5*p**4 - 6*p + 30)
+    assert G3.mul(G4) == expect
+    assert G3.mul(s) == TransferFunction(p*s, p**4 - 6)
+    assert G4.mul(p**-1) == TransferFunction(p + 4, p**2 - 5*p)
+
+    G5 = TransferFunction(s + 6, s - 5)
+    G6 = TransferFunction(s + 3, s + 1)
+    assert G5.div(G6) == TransferFunction(s**2 + 7*s + 6, s**2 - 2*s - 15)
+    assert G5.div(2) == TransferFunction(s + 6, 2*s - 10)
```

Comment:
There is no need for special mul and div methods. Just use `tf1*tf2` and `tf2/tf1`.